Dockerfile

```diff
@@ -1,16 +1,32 @@
-FROM i386/debian:9
+FROM i386/debian:stable AS builder
+RUN useradd -md /opt/geneshift geneshift
+RUN apt-get update && apt-get upgrade -y
+RUN apt-get install -y curl ca-certificates
 
 ENV DUMB_INIT_VERSION=1.2.2 \
-    GENESHIFT_VERSION=1243 \
-    GOSU_VERSION=1.11
+    GENESHIFT_VERSION=1264
 
-COPY build.sh /usr/local/bin/
-RUN set -ex \
- && bash /usr/local/bin/build.sh
+RUN apt-get install curl ca-certificates -y
+
+RUN curl -sSfLo /tmp/Geneshift${GENESHIFT_VERSION}.tar.gz "https://www.geneshift.net/downloads/Geneshift${GENESHIFT_VERSION}.tar.gz"
+RUN curl -sSfLo /usr/bin/dumb-init "https://github.com/Yelp/dumb-init/releases/download/v${DUMB_INIT_VERSION}/dumb-init_${DUMB_INIT_VERSION}_amd64"
+RUN tar -vxzf "/tmp/Geneshift${GENESHIFT_VERSION}.tar.gz" -C /opt/geneshift
+RUN chmod a+x /usr/bin/dumb-init
+
+FROM i386/debian:stable
+RUN useradd -md /opt/geneshift geneshift
+RUN apt-get update && apt-get upgrade -y
+
+COPY --from=builder /usr/bin/dumb-init /usr/bin/
+COPY --from=builder --chown=geneshift /opt/geneshift/* /opt/geneshift/
+ADD run.sh /usr/local/bin/
 
 VOLUME /config
 
 EXPOSE 11235/udp
 
-ADD run.sh /usr/local/bin/
-ENTRYPOINT /usr/local/bin/run.sh
+USER geneshift
+
+ENTRYPOINT ["/usr/bin/dumb-init", "--"]
+
+CMD ["/usr/local/bin/run.sh"]
```

Comment:
This now has 11 layers instead of the original 8 layers so it increases the number of layers instead of decreasing